### PR TITLE
Create report-data-provider.dart

### PR DESCRIPTION
**Pull request by million-t**
this is a data provider for "report" feature for a vehicle management flutter app using bloc.
Take some time, and review this.

### CODE REVIEW

**Comment by AnaniyaT** [comment]
Good overall. But I don't see a single heap in your implementation. ADD HEAPS IN YOUR CODE. HEAPS ARE GREAT!! YOU NEED HEAPS!!
               - Meraf (probably)

File: report-data-provider.dart [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import 'dart:convert';
+import '../../util/serverIp.dart';
+import 'package:http/http.dart' as http;
+
+import '../models/report_model.dart';
+
+class ReportDataProvider {
+  static final String _baseUrl = "${Server().ip}/report";
```
Comment (meraf00) [inline report-data-provider.dart:8]: Good job, encapsulating low level detail using data provider, also collecting environmental settings in one place like (util/serverIp.dart)

File: report-data-provider.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  Future<Report> create(Report report, String token) async {
+    final http.Response response = await http.post(Uri.parse(_baseUrl),
```
Comment (meraf00) [inline report-data-provider.dart:11]: Improvement:
currently we are directly using http client to do requests, you might consider using [Facade pattern](https://kazlauskas.dev/flutter-design-patterns-7-facade/) for ease of testing or incase we want to change client later.

File: report-data-provider.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    final response = await http.get(Uri.parse("$_baseUrl/$id"));
+
+    if (response.statusCode == 200) {
+      return Report.fromJson(jsonDecode(response.body));
+    } else {
+      throw Exception("Fetching Report failed");
+    }
+  }
+
+  Future<List<Report>> getAllByDriver(String token) async {
```
Comment (meraf00) [inline report-data-provider.dart:44]: Improvement:
Pagination - we are fetching all drivers at once which might cause performance issue for large dataset, we might need to consider pagination.

File: report-data-provider.dart [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+  Future<void> delete(String id, String token) async {
+    final response = await http.delete(Uri.parse("$_baseUrl/$id"),
+        headers: <String, String>{"Authorization": token});
+    if (response.statusCode != 204) {
+      throw Exception("Field to delete the Reports");
+    }
+    return;
+  }
+}
```
Comment (meraf00) [inline report-data-provider.dart:98]: Liked your clean and easy to follow code. Good job!

**Review by meraf00 — COMMENTED** [review]
Liked your clean and easy to follow code. Good function and variable naming. Good job!

**Pull request merged**